core/Command/Background/JobWorker.php

```diff
@@ -0,0 +1,154 @@
+use OCP\ITempManager;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class JobWorker extends JobBase {
+	protected function configure(): void {
+		parent::configure();
+
+		$this
+			->setName('background-job:worker')
+			->setDescription('Run a background job worker')
+			->addArgument(
+				'job-classes',
+				InputArgument::OPTIONAL,
+				'The classes of the jobs to look for in the database, comma-separated'
+			)
+			->addOption(
+				'once',
+				null,
+				InputOption::VALUE_NONE,
+				'Only execute the worker once (as a regular cron execution would do it)'
+			)
+			->addOption(
+				'interval',
+				'i',
+				InputOption::VALUE_OPTIONAL,
+				'Interval in seconds in which the worker should repeat already processed jobs (set to 0 for no repeat)',
+				5
+			)
+		;
+	}
+
+	protected function execute(InputInterface $input, OutputInterface $output): int {
+		$jobClassesString = $input->getArgument('job-classes');
+		// only keep non-empty strings
+		$jobClasses = $jobClassesString === null
+			? null
+			: array_filter(
+				explode(',', $jobClassesString),
+				static function (string $jobClass) {
+					return strlen($jobClass) > 0;
+				}
+			);
+
+		if ($jobClasses !== null) {
+			// no class
+			if (count($jobClasses) === 0) {
+				$output->writeln('<error>Invalid job class list supplied</error>');
+				return 1;
+			}
+
+			// at least one invalid class
+			foreach ($jobClasses as $jobClass) {
+				if (!class_exists($jobClass)) {
+					$output->writeln('<error>Invalid job class: ' . $jobClass . '</error>');
+					return 1;
+				}
+			}
+		}
+
+		while (true) {
+			// Handle canceling of the process
+			try {
+				$this->abortIfInterrupted();
+			} catch (InterruptedException $e) {
+				$output->writeln('<info>Background job worker stopped</info>');
+				break;
+			}
+
+			$this->printSummary($input, $output);
+
+			usleep(50000);
+			$job = $this->jobList->getNext(false, $jobClasses);
+			if (!$job) {
+				if ($input->getOption('once') === true) {
+					$output->writeln('No job of classes ' . $jobClassesString . ' is currently queued', OutputInterface::VERBOSITY_VERBOSE);
+					$output->writeln('Exiting...', OutputInterface::VERBOSITY_VERBOSE);
+					break;
+				}
+
+				$output->writeln('Waiting for new jobs to be queued', OutputInterface::VERBOSITY_VERBOSE);
+				// Re-check interval for new jobs
+				sleep(1);
+				continue;
+			}
+
+			$output->writeln('Running job ' . get_class($job) . ' with ID ' . $job->getId());
+
+			if ($output->isVerbose()) {
+				$this->printJobInfo($job->getId(), $job, $output);
+			}
+
+			$job->start($this->jobList);
+
+			$output->writeln('Job ' . $job->getId() . ' has finished', OutputInterface::VERBOSITY_VERBOSE);
+
+			// clean up after unclean jobs
+			\OC_Util::tearDownFS();
+			\OC::$server->get(ITempManager::class)->clean();
```

Comment:
```suggestion
			\OCP\Server::get(ITempManager::class)->clean();
```
But it could even be kept in a var, or injected in constructor?